polynote-spark/src/main/scala/polynote/kernel/interpreter/python/PySparkInterpreter.scala

```diff
@@ -102,8 +102,11 @@ class PySparkInterpreter(
        |    spark_python = os.path.join(spark_home, "python")
        |    sys.path.insert(1, spark_python)
        |    import glob
-       |    py4j_path = glob.glob(os.path.join(spark_home, 'python', 'lib', 'py4j-*.zip'))[0]  # we want to use the py4j distributed with pyspark
-       |    sys.path.insert(1, py4j_path)
+       |    try:
+       |        py4j_path = glob.glob(os.path.join(spark_home, 'python', 'lib', 'py4j-*.zip'))[0]  # we want to use the py4j distributed with pyspark
+       |        sys.path.insert(1, py4j_path)
+       |    except IndexError as err:
+       |        raise IndexError('No py4j found using SPARK_HOME: ' + spark_home)
```

Comment:
+1, definitely a better message!

Curious about how you ran into this error in the first place though - some spark distribution without py4j?

Reply:
Oh, I missed https://github.com/polynote/polynote/pull/1201#issuecomment-913898133

Still, curious about that error.. weird.